File: ctopy.py
#!/usr/bin/env python3

import socket
import os
import sys
from time import sleep

port = 2500

s = socket.socket(socket.AF_INET,socket.SOCK_STREAM);

s.bind(('',port))
print("Parent: Socket binded to port " + str(port))

s.listen(3)

sleep(1)
while True:
    c, addr = s.accept()
    print("Parent: Got connection from" + str(addr))

    while True:
        message = c.recv(1024).decode()
        if message != "":
            print(message)
        else:
            print("Parent: Closing connection from" + str(addr))
            c.close()
            break

    #cont = input("Continue receiving connections?: {y/n}")
    #if cont == "n":
    #    break

s.close()
exit(0)

File: ctopy.c
#include <sys/types.h>
#include <sys/socket.h>
#include <unistd.h>
#include <stdio.h>
#include <netinet/in.h>
#include <stdlib.h>

#define PORT 2500

int main(void){
    struct sockaddr_in addrs;
    int adrlen = sizeof(addrs);
    addrs.sin_family = AF_INET;
    addrs.sin_port = htons(PORT);
    addrs.sin_addr.s_addr = INADDR_ANY;
    int client_fd = socket(AF_INET,SOCK_STREAM,0);
    int conn = connect(client_fd,(struct sockaddr*)&addrs,sizeof(addrs));
    while(0 < 1){
        char buffer[1024];
        printf("What to send?\n");
        char c;
        int i = 0;
        scanf("%c",&c);
        while(c != '\n'){
            buffer[i] = c;
            i++;
            scanf("%c",&c);
        }
        buffer[i] = '\0';
        if (i == 0){
            printf("Client closing connection\n");
            close(client_fd);
            exit(0);
        }else{
            while(i < 1024){
                buffer[i] = '\0';
            }
            int sendd = send(client_fd,buffer,sizeof(buffer),0);
            printf("Child sent %s\n",buffer);
        }
    }
}
